### PR TITLE
build: moving subset lb into extensions

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: inline

### CODE REVIEW

**Comment by alyssawilk** [comment]
hey @wbpcode want to get your thoughts on this.  I know it isn't furthering the plan you had of making subset lb a thread aware load balancer, but I do think it makes a follow-up PR switching the factory type the way you mentioned a bit simpler, and it advances my goal of getting more code moved out of the core build.  If you agree it doesn't make things harder for you I'm going to  land this and a series of other similar clean ups for LB code in the LB directories.

**Comment by wbpcode** [comment]
Thanks for advancing this work. This pr is completely ok for me. And the code moving is also something I originally intend to do.
So, just go ahead.

---
Just for discussion:
Seems that you didn't introduce any new proto api for the subset lb.
It's more like a code clean-up rather then make the subset lb as an configable extension in `load_balancing_policy`.

**Comment by alyssawilk** [comment]
@zuercher ping?

File: CODEOWNERS [hunk 1]
```diff
@@ -293,6 +293,7 @@ extensions/filters/http/oauth2 @derekargueta @snowp
 /*/extensions/load_balancing_policies/round_robin @wbpcode @UNOWNED
 /*/extensions/load_balancing_policies/ring_hash @wbpcode @UNOWNED
 /*/extensions/load_balancing_policies/maglev @wbpcode @UNOWNED
+/*/extensions/load_balancing_policies/subset @wbpcode @zuecher
```
Comment (zuercher) [inline CODEOWNERS:296]: should be @zuercher

**Review by zuercher — COMMENTED** [review]
Looks good, just has the wrong handle in the CODEOWNERS

**Review by zuercher — APPROVED** [review]

**Pull request merged**